app/addons/documents/revision-tree/component.react.jsx

```diff
@@ -0,0 +1,338 @@
+define([
+  "app",
+  "api",
+  "react",
+  'addons/documents/revision-tree/stores',
+  'addons/documents/revision-tree/actions',
+],
+
+function (app, FauxtonAPI, React, Stores) {
+
+  var store = Stores.revTreeStore;
+  var lineObjs = [];
+  var nodeObjs = [];
+  var textObjs = [];
+  var grid = 100;
+  var scale = 7;
+  var r = 15;
+
+  var LinesBox = React.createClass({
+    render: function () {
+      return (
+        <g>
+          {
+            this.props.data.map(function (element, i) {
+              return <Line key={i} data={element}/>;
+            })
+          }
+        </g>
+      );
+    }
+  });
+
+  var CirclesBox = React.createClass({
+
+    render: function () {
+      return (
+        <g>
+          {
+            this.props.data.map(function (element, i) {
+              return <Circle key={i} data={element}/>;
+            })
+          }
+        </g>
+      );
+    }
+  });
+
+  var Circle = React.createClass({
+
+    render: function () {
+      var cx = this.props.data.x;
+      var cy = this.props.data.y;
+      var radius = this.props.data.radius;
+      var classVal = this.props.data.class;
+
+      return (
+        <circle cx={cx} cy={cy} r={radius} className={classVal}>{this.props.children}</circle>
+      );
+    }
+  });
+
+  var Line = React.createClass({
+
+    render: function () {
+      var x1 = this.props.data.x;
+      var y1 = this.props.data.y;
+      var x2 = this.props.data.nextX;
+      var y2 = this.props.data.nextY;
+
+      return (
+        <line x1={x1} y1={y1} x2={x2} y2={y2}>{this.props.children}</line>
+      );
+    }
+  });
+
+  var TextsBox = React.createClass({
+
+    render: function () {
+      return (
+        <g>{this.props.children}</g>
+      );
+    }
+  });
+
+  var SVGComponent = React.createClass({
+
+    render: function () {
+      return (
+        <svg height="500" width="500">{this.props.children}
+        </svg>
+      );
+    }
+  });
+
+  var Box = React.createClass({
+
+    render: function () {
+      return (
+        <div className="visualizeRevTree">
+          {this.props.children}
+        </div>
+      );
+    }
+  });
+
+  var Text = React.createClass({
+
+    render: function () {
+      var styleVals = {
+        left: this.props.data.stLeft,
+        top: this.props.data.stTop
+      };
+
+      return (
+          <div className="box" style={styleVals} short={this.props.data.short} long={this.props.data.long}>
+            <p>{this.props.data.short}</p>
+            {this.props.children}
+          </div>
+      );
+    }
+  });
+
+  var draw = function (paths, deleted, winner, minUniq) {
+    var maxX = grid;
+    var maxY = grid;
+    var levelCount = []; // numer of nodes on some level (pos)
+    lineObjs.length = 0;
+    nodeObjs.length = 0;
+    textObjs.length = 0;
```

Comment:
i think we should not depend on global state and just initialize new arrays, instead of resetting them